### PR TITLE
[CHA-14] Added support for pinning, archiving channels and partial member update

### PR DESCRIPTION
**Pull request by totalimmersion**
https://linear.app/stream/issue/CHA-14/members-custom-data-go-sdk

### CODE REVIEW

File: channel.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ type ChannelMember struct {
 	InviteAcceptedAt *time.Time `json:"invite_accepted_at,omitempty"`
 	InviteRejectedAt *time.Time `json:"invite_rejected_at,omitempty"`
 	Role             string     `json:"role,omitempty"`
+	ArchivedAt       *time.Time `json:"archived_at,omitempty"`
```
Comment (akupila) [inline channel.go:30]:
```suggestion
```

**Review by akupila — COMMENTED** [review]
~This only adds support for pinning, so we should probably exclude `ArchivedAt` (or also add that here, as it's very similar)~

File: channel.go (hunk 1 above)
Comment (totalimmersion) [reply channel.go:30]: it's still work in progress, i'll convert it to draft

File: channel_test.go [hunk 2]
```diff
@@ -647,6 +683,108 @@ func TestChannel_Mute_Unmute(t *testing.T) {
...
+	ctx := context.Background()
+	users := randomUsers(t, c, 5)
+
+	members := make([]string, 0, len(users))
+	for i := range users {
+		members = append(members, users[i].ID)
+	}
+	ch := initChannel(t, c, members...)
+
+	//pin the channel
```
Comment (akupila) [inline channel_test.go:748]:
```suggestion
	//archive the channel
```

File: channel_test.go [hunk 3]
```diff
@@ -647,6 +683,108 @@ func TestChannel_Mute_Unmute(t *testing.T) {
...
+	channels := queryChannResp.Channels
+	require.NoError(t, err, "query archived channel")
+	require.Len(t, channels, 1)
+	require.Equal(t, channels[0].CID, ch.CID)
+
+	member, err = ch.Unarchive(ctx, users[0].ID)
+	require.NoError(t, err, "unarchive channel")
+	require.Nil(t, member.ChannelMember.ArchivedAt)
+
+	// query for pinned the channel
```
Comment (akupila) [inline channel_test.go:773]:
```suggestion
	// query for the archived channel
```

**Review by akupila — DISMISSED** [review]

**Pull request merged**